// @openrouter/sdk/esm/sdk/workspaces.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 72b45379c8d9
 */
import { workspacesBulkAddMembers } from "../funcs/workspacesBulkAddMembers.js";
import { workspacesBulkRemoveMembers } from "../funcs/workspacesBulkRemoveMembers.js";
import { workspacesCreate } from "../funcs/workspacesCreate.js";
import { workspacesDelete } from "../funcs/workspacesDelete.js";
import { workspacesGet } from "../funcs/workspacesGet.js";
import { workspacesList } from "../funcs/workspacesList.js";
import { workspacesUpdate } from "../funcs/workspacesUpdate.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
import { unwrapResultIterator } from "../types/operations.js";
export class Workspaces extends ClientSDK {
    /**
     * List workspaces
     *
     * @remarks
     * List all workspaces for the authenticated user. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async list(request, options) {
        return unwrapResultIterator(workspacesList(this, request, options));
    }
    /**
     * Create a workspace
     *
     * @remarks
     * Create a new workspace for the authenticated user. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async create(request, options) {
        return unwrapAsync(workspacesCreate(this, request, options));
    }
    /**
     * Delete a workspace
     *
     * @remarks
     * Delete an existing workspace. The default workspace cannot be deleted. Workspaces with active API keys cannot be deleted. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async delete(request, options) {
        return unwrapAsync(workspacesDelete(this, request, options));
    }
    /**
     * Get a workspace
     *
     * @remarks
     * Get a single workspace by ID or slug. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async get(request, options) {
        return unwrapAsync(workspacesGet(this, request, options));
    }
    /**
     * Update a workspace
     *
     * @remarks
     * Update an existing workspace by ID or slug. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async update(request, options) {
        return unwrapAsync(workspacesUpdate(this, request, options));
    }
    /**
     * Bulk add members to a workspace
     *
     * @remarks
     * Add multiple organization members to a workspace. Members are assigned the same role they hold in the organization. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async bulkAddMembers(request, options) {
        return unwrapAsync(workspacesBulkAddMembers(this, request, options));
    }
    /**
     * Bulk remove members from a workspace
     *
     * @remarks
     * Remove multiple members from a workspace. Members with active API keys in the workspace cannot be removed. [Management key](/docs/guides/overview/auth/management-api-keys) required.
     */
    async bulkRemoveMembers(request, options) {
        return unwrapAsync(workspacesBulkRemoveMembers(this, request, options));
    }
}
