// @openrouter/sdk/esm/models/contentfilterbuiltinentry.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: a4c9b1effbc1
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ContentFilterBuiltinAction$inboundSchema, ContentFilterBuiltinAction$outboundSchema, } from "./contentfilterbuiltinaction.js";
import { ContentFilterBuiltinSlug$inboundSchema, ContentFilterBuiltinSlug$outboundSchema, } from "./contentfilterbuiltinslug.js";
/** @internal */
export const ContentFilterBuiltinEntry$inboundSchema = z.object({
    action: ContentFilterBuiltinAction$inboundSchema,
    label: z.string().optional(),
    slug: ContentFilterBuiltinSlug$inboundSchema,
});
/** @internal */
export const ContentFilterBuiltinEntry$outboundSchema = z.object({
    action: ContentFilterBuiltinAction$outboundSchema,
    label: z.string().optional(),
    slug: ContentFilterBuiltinSlug$outboundSchema,
});
export function contentFilterBuiltinEntryToJSON(contentFilterBuiltinEntry) {
    return JSON.stringify(ContentFilterBuiltinEntry$outboundSchema.parse(contentFilterBuiltinEntry));
}
export function contentFilterBuiltinEntryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ContentFilterBuiltinEntry$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ContentFilterBuiltinEntry' from JSON`);
}
