// @openrouter/sdk/esm/models/outputapplypatchservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 761046712899
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ToolCallStatus$inboundSchema, ToolCallStatus$outboundSchema, } from "./toolcallstatus.js";
export const OutputApplyPatchServerToolItemType = {
    OpenrouterApplyPatch: "openrouter:apply_patch",
};
/** @internal */
export const OutputApplyPatchServerToolItemType$inboundSchema = z.enum(OutputApplyPatchServerToolItemType);
/** @internal */
export const OutputApplyPatchServerToolItemType$outboundSchema = OutputApplyPatchServerToolItemType$inboundSchema;
/** @internal */
export const OutputApplyPatchServerToolItem$inboundSchema = z.object({
    filePath: z.string().optional(),
    id: z.string().optional(),
    patch: z.string().optional(),
    status: ToolCallStatus$inboundSchema,
    type: OutputApplyPatchServerToolItemType$inboundSchema,
});
/** @internal */
export const OutputApplyPatchServerToolItem$outboundSchema = z.object({
    filePath: z.string().optional(),
    id: z.string().optional(),
    patch: z.string().optional(),
    status: ToolCallStatus$outboundSchema,
    type: OutputApplyPatchServerToolItemType$outboundSchema,
});
export function outputApplyPatchServerToolItemToJSON(outputApplyPatchServerToolItem) {
    return JSON.stringify(OutputApplyPatchServerToolItem$outboundSchema.parse(outputApplyPatchServerToolItem));
}
export function outputApplyPatchServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputApplyPatchServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputApplyPatchServerToolItem' from JSON`);
}
