// @openrouter/sdk/esm/models/sttrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3be8453c0517
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { ProviderOptions$outboundSchema, } from "./provideroptions.js";
import { STTInputAudio$outboundSchema, } from "./sttinputaudio.js";
/** @internal */
export const STTRequestProvider$outboundSchema = z.object({
    options: ProviderOptions$outboundSchema.optional(),
});
export function sttRequestProviderToJSON(sttRequestProvider) {
    return JSON.stringify(STTRequestProvider$outboundSchema.parse(sttRequestProvider));
}
/** @internal */
export const STTRequest$outboundSchema = z.object({
    inputAudio: STTInputAudio$outboundSchema,
    language: z.string().optional(),
    model: z.string(),
    provider: z.lazy(() => STTRequestProvider$outboundSchema).optional(),
    temperature: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        inputAudio: "input_audio",
    });
});
export function sttRequestToJSON(sttRequest) {
    return JSON.stringify(STTRequest$outboundSchema.parse(sttRequest));
}
